### PR TITLE
Release v2.19.0

### PR DESCRIPTION
**Pull request by zhiqiangjin**

### CODE REVIEW

**Review by mlin-aviatrix — DISMISSED** [review]

**Review by shuoliugit — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,96 @@
...
+### Notes:
+- Supported Controller version: **UserConnect-6.4.2416**
+- Supported Terraform version: **v0.12.x**, **v0.13.x** and **v0.14.x**
+
+### Features:
+1. Implemented support of following attributes in **aviatrix_spoke_gateway**:
+  - ``enable_private_vpc_default_route``
+  - ``enable_skip_public_route_table_update``
+  - ``enable_auto_advertise_s2c_cidrs``
+2. Implemented support of associating a controller with a CoPilot instance:
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 2 "... , allowing user login without a username and password:"

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,96 @@
-## 2.18.2 (Unreleased)
+## 2.19.0 (Unreleased)
+### Notes:
+- Supported Controller version: **UserConnect-6.4.2416**
+- Supported Terraform version: **v0.12.x**, **v0.13.x** and **v0.14.x**
+
+### Features:
+1. Implemented support of following attributes in **aviatrix_spoke_gateway**:
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 1. Implemented support for the following attributes in ...

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,96 @@
...
+  - ``enable_private_vpc_default_route``
+  - ``enable_skip_public_route_table_update``
+  - ``enable_auto_advertise_s2c_cidrs``
+2. Implemented support of associating a controller with a CoPilot instance:
+  - New resource **aviatrix_copilot_association**
+3. Implemented support for enabling event triggered HA for Site2Cloud type connection resources:
+  - New attribute ``enable_event_triggered_ha`` in **aviatrix_site2cloud**, **aviatrix_transit_external_device_conn**, **aviatrix_vgw_conn** and **aviatrix_device_transit_gateway_attachment**
+4. Implemented support for GCP FireNet with Fortinet and CheckPoint firewall vendors
+5. Implemented support for Insane Mode over Internet with following attributes in **aviatrix_transit_gateway_peering**:
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 5. Implemented Insane Mode Support over Public Network for Transit Peering in **aviatrix_transit_gateway_peering**:

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,96 @@
...
+  - ``enable_insane_mode_encryption_over_internet``
+  - ``tunnel_count``
+6. Implemented support for attaching a managed CloudN device to an **aviatrix_transit_gateway**:
+  - New resource **aviatrix_cloudn_transit_gateway_attachment**
+7. Implemented support for setting approved CIDRs in **aviatrix_transit_external_device_conn**:
+  - New attribute ``approved_cidrs``
+8. Implemented support for remote syslog profile name:
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 8. Implemented support for adding profile names to Remote Syslog configs:

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,4 +1,96 @@
...
+  - New attribute ``name`` in **aviatrix_remote_syslog**
+9. Implemented new resources to decouple ``security_domains`` out of **aviatrix_aws_tgw**:
+  - **aviatrix_aws_tgw_security_domain**
+  - **aviatrix_aws_tgw_security_domain_connection**
+10. Implemented support for intra domain inspection:
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 10. Implemented support for TGW intra-domain inspection:

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,4 +1,96 @@
...
+  - New resource **aviatrix_aws_tgw_intra_domain_inspection**
+11. Implemented support for multi-tier transit feature:
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 11. Implemented support for Multi-Tier Transit feature:

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,4 +1,96 @@
...
+  - New attribute ``enable_multi_tier_transit`` in **aviatrix_transit_gateway** and data source
+12. Implemented support for Alibaba Cloud in resources **aviatrix_account**, **aviatrix_vpc**, **aviatrix_gateway**, **aviatrix_transit_gateway** and **aviatrix_spoke_gateway** and data sources
+13. Implemented support for disabling sending exception emails to Aviatrix:
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 13. Implemented support for enabling/disabling Controller from sending exception emails to Aviatrix:

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,4 +1,96 @@
...
+  - New resource **aviatrix_controller_email_exception_notification_config**
+14. Implemented support for updating certificate domain:
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 14. Implemented support for updating Controller's certificate domain, required for Aviatrix China Solution:

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,4 +1,96 @@
-## 2.18.2 (Unreleased)
+## 2.19.0 (Unreleased)
+### Notes:
+- Supported Controller version: **UserConnect-6.4.2416**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: version will likely change before we release. At the moment of writing, it is 6.4.2433

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,4 +1,96 @@
...
+10. Implemented support for intra domain inspection:
+  - New resource **aviatrix_aws_tgw_intra_domain_inspection**
+11. Implemented support for multi-tier transit feature:
+  - New attribute ``enable_multi_tier_transit`` in **aviatrix_transit_gateway** and data source
+12. Implemented support for Alibaba Cloud in resources **aviatrix_account**, **aviatrix_vpc**, **aviatrix_gateway**, **aviatrix_transit_gateway** and **aviatrix_spoke_gateway** and data sources
+13. Implemented support for disabling sending exception emails to Aviatrix:
+  - New resource **aviatrix_controller_email_exception_notification_config**
+14. Implemented support for updating certificate domain:
+  - New resource **aviatrix_controller_cert_domain_config**
+15. Implemented support for TGW segmentation for Egress:
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 15. Implemented support for TGW segmentation for Egress in TGW FireNet workflows:

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,4 +1,96 @@
...
+  - New attribute ``tgw_segmentation_for_egress_enabled`` in **aviatrix_firenet** and data source
+16. Implemented support for OCI FireNet
+17. Implemented support for Egress FireNet route injection:
+  - New attribute ``egress_static_cidrs`` in **aviatrix_firenet** and data source
+18. Implemented support for launching user provided AMI for FireNet:
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 18. Implemented custom AMI support for Firewall instance, allowing customers to launch special images provided by firewall vendors:

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,4 +1,96 @@
...
+  - New attribute ``firewall_image_id`` in **aviatrix_firewall_instance**
+19. Implemented support for optional tunnel ip addresses with following attributes in **aviatrix_site2cloud**:
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 19. Implemented support for setting optional tunnel IP address with the following attributes in **aviatrix_site2cloud**:

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,4 +1,96 @@
...
+  - ``local_tunnel_ip``
+  - ``remote_tunnel_ip``
+  - ``backup_local_tunnel_ip``
+  - ``backup_remote_tunnel_ip``
+20. Implemented support for phase 1 remote identifier for Site2Cloud:
+  - New attribute ``phase1_remote_identifier`` in **aviatrix_site2cloud**
+21. Implemented support for AzureGov, AWSChina and AzureChina clouds in resources **aviatrix_account**, **aviatrix_vpc**, **aviatrix_gateway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
+22. Implemented support for creating an **aviatrix_vpc** in Azure with an existing resource group ``resource_group``
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: remove the non-formatted "resource group". looks repetitive

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,4 +1,96 @@
...
+23. Implemented support for separate IAM role and policy for gateways in AWS **aviatrix_account** with new attributes:
+  - ``aws_gateway_role_app``
+  - ``aws_gateway_role_ec2``
+24. Implemented support for IPSec tunnel down detection time in **aviatrix_gatewway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**:
+  - New attribute ``tunnel_detection_time``
+25. Implemented support for **aviatrix_site2cloud** connections to have a single public IP with two gateways:
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 25. Implemented single public IP failover support for **aviatrix_site2cloud** connections:

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,4 +1,96 @@
...
+  - New attribute ``single_ip_ha``
+26. Implemented support for enabling auditing in **aviatrix_account**:
+  - New attribute ``audit_account``
+27. Implemented support for BGP max AS limit controller configuration:
+  - New resource **aviatrix_controller_bgp_max_as_limit_config**
+28. Implemented support for OCI insane mode in **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 28. Implemented OCI transit Insane Mode support in **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,4 +1,96 @@
...
+### Enhancements:
+1. Added following attributes in **aviatrix_account** data source:
+  - ``gcloud_project_id``
+  - ``arm_subscription_id``
+  - ``awsgov_account_number``
+  - ``awsgov_access_key``
+2. Optimized state refresh performance for **aviatrix_gateway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
+3. Added support for Fortinet Fortigate in **data_source_aviatrix_firenet_vendor_integration**
+4. Added computed value ``tgw_id`` in **aviatrix_aws_tgw**
+5. Added new map type attribute ``tags`` to replace ``tag_list`` in **aviatrix_gatewway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: typo in **aviatrix_gateway**

- also, is it ok to have this point under enhancements, as this is also a duplicate as a deprecation point # 2 ?

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,4 +1,96 @@
...
+6. Changed ``aws_access_key`` and ``aws_gov_access_key`` in **aviatrix_acount** to be sensitive values
+
+### Bug Fixes:
+1. Fixed an edge case in **aviatrix_gateway** that could cause the provider to crash when refreshing the resource
+2. Fixed **aviatrix_transit_gateway_peering** to allow setting duplicate AS Numbers in the ``prepend_as_path1`` and ``prepend_as_path2`` attributes
+3. Fixed **aviatrix_fqdn** to not remove ``domain_names`` after importing the resource with ``manage_domain_names`` set to false
+4. Fixed reordering issue for ``security_domains`` in **aviatrix_aws_tgw**
+5. Fixed issue where disabling transit firenet and downsizing the gateway can't be done in one run
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: ...Transit FireNet option and downsizing the gateway cannot be completed in one Terraform operation

File: CHANGELOG.md [hunk 18]
```diff
@@ -1,4 +1,96 @@
...
+6. Fixed issue where enabling HA for **aviatrix_gateway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway** with insane mode enabled errors out
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Fixed issue where enabling HA for Insane Mode **aviatrix_gateway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway** will cause Terraform to error out

File: CHANGELOG.md [hunk 19]
```diff
@@ -1,4 +1,96 @@
...
+7. Fixed issue where disabling transit firenet and egress transit firenet can't be done in one run in **aviatrix_transit_gateway**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Fixed issue where disabling Transit FireNet and Egress Transit FireNet options cannot be completed in one Terraform operation in **aviatrix_transit_gateway**

File: CHANGELOG.md [hunk 20]
```diff
@@ -1,4 +1,96 @@
...
+  - ``backup_local_tunnel_ip``
+  - ``backup_remote_tunnel_ip``
+20. Implemented support for phase 1 remote identifier for Site2Cloud:
+  - New attribute ``phase1_remote_identifier`` in **aviatrix_site2cloud**
+21. Implemented support for AzureGov, AWSChina and AzureChina clouds in resources **aviatrix_account**, **aviatrix_vpc**, **aviatrix_gateway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
+22. Implemented support for creating an **aviatrix_vpc** in Azure with an existing resource group ``resource_group``
+23. Implemented support for separate IAM role and policy for gateways in AWS **aviatrix_account** with new attributes:
+  - ``aws_gateway_role_app``
+  - ``aws_gateway_role_ec2``
+24. Implemented support for IPSec tunnel down detection time in **aviatrix_gatewway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**:
```
Comment (CyrusJavan) [inline CHANGELOG.md]:
```suggestion
24. Implemented support for IPSec tunnel down detection time in **aviatrix_gateway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**:
```

File: CHANGELOG.md [hunk 21]
```diff
@@ -1,4 +1,96 @@
...
+3. Fixed **aviatrix_fqdn** to not remove ``domain_names`` after importing the resource with ``manage_domain_names`` set to false
+4. Fixed reordering issue for ``security_domains`` in **aviatrix_aws_tgw**
+5. Fixed issue where disabling transit firenet and downsizing the gateway can't be done in one run
+6. Fixed issue where enabling HA for **aviatrix_gateway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway** with insane mode enabled errors out
+7. Fixed issue where disabling transit firenet and egress transit firenet can't be done in one run in **aviatrix_transit_gateway**
+
+### Deprecations:
+1. Deprecated the in-line attributes ``security_domains``, ``security_domain_name``, ``connected_domains``, ``aviatrix_firewall``, ``native_egress`` and ``native_firewall`` in **aviatrix_aws_tgw**. Please use the standalone resources **aviatrix_aws_tgw_security_domain** and **aviatrix_aws_tgw_security_domain_connection** instead
+2. Deprecated ``tag_list`` in **aviatrix_gatewway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway** for new map type attribute ``tags``
```
Comment (CyrusJavan) [inline CHANGELOG.md]:
```suggestion
2. Deprecated ``tag_list`` in **aviatrix_gateway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway** for new map type attribute ``tags``
```

Comment (CyrusJavan) [inline CHANGELOG.md]: Deprecated ``tag_list`` in **aviatrix_gateway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**. Please use map type attribute ``tags`` instead

**Review by CyrusJavan — DISMISSED** [review]

File: CHANGELOG.md [hunk 22]
```diff
@@ -1,4 +1,127 @@
...
+2. Implemented support for AzureGov, AWSChina and AzureChina clouds in resource **aviatrix_gateway**
+3. Implemented support for IPSec tunnel down detection time in **aviatrix_gateway**:
+  - New attribute ``tunnel_detection_time``
+
+#### Multi-Cloud Transit
+1. Implemented support for the following attributes in **aviatrix_spoke_gateway**:
+  - ``enable_private_vpc_default_route``
+  - ``enable_skip_public_route_table_update``
+  - ``enable_auto_advertise_s2c_cidrs``
+2. Implemented support for enabling event triggered HA for Site2Cloud type connection resources:
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: "Implemented support for enabling Event Triggered HA ..."

File: CHANGELOG.md [hunk 23]
```diff
@@ -1,4 +1,127 @@
...
+8. Implemented support for AzureGov, AWSChina and AzureChina clouds in resources **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
+9. Implemented support for IPSec tunnel down detection time in **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**:
+  - New attribute ``tunnel_detection_time``
+10. Implemented OCI transit Insane Mode support in **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
+11. Implemented support for ``enable_egress_transit_firenet`` for Azure and OCI in **aviatrix_transit_gateway**
+12. Implemented support for phase 1 remote identifier in **aviatrix_transit_external_device_conn**:
+  - New attribute ``phase1_remote_identifier``
+
+#### Settings
+1. Implemented support of associating a controller with a CoPilot instance, allowing user login without a username and password:
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: "Implemented support for associating a Controller with a CoPilot instance..."

File: CHANGELOG.md [hunk 24]
```diff
@@ -1,4 +1,127 @@
-## 2.18.2 (Unreleased)
+## 2.19.0 (Unreleased)
+### Notes:
+- Supported Controller version: **UserConnect-6.4.2433**
```
Comment (inc1t3Ful) [inline CHANGELOG.md:3]: this will still need to be updated, once we know the official version

**Review by inc1t3Ful — APPROVED** [review]

**Pull request merged**